### PR TITLE
PS-7116 MyRocks: Port fixes of Index Condition Pushdown (8.0)

### PR DESCRIPTION
**Pull request by inikep**
Fix Index Condition Pushdown (ICP) issues in MyRocks with Percona Server 8.0 that were introduced when porting MyRocks to Percona Server 8.0. End range needs to be set for bloom filter to take effect.

This patch ports the following commits:
https://github.com/facebook/mysql-5.6/commit/5dfbcb248c7cdc6c2eaa7c53254befc7d29695d4
https://github.com/facebook/mysql-5.6/commit/52585179a20a00fc9033a7d4f88be6e89517eef0

### CODE REVIEW

**Comment by inikep** [comment]
Tested at https://ps80.cd.percona.com/job/percona-server-8.0-param/601/

File: sql/handler.cc [hunk 1]
```diff
@@ -7835,6 +7829,8 @@ int handler::compare_key_in_buffer(const uchar *buf) const {
   // Reset the field offsets.
   if (diff != 0) move_key_field_offsets(end_range, range_key_part, -diff);
 
+  if (range_scan_direction == RANGE_SCAN_DESC) cmp = -cmp;
```
Comment (george-lorch) [inline sql/handler.cc:7833]: Please add a comment noting that this change is necessary for MyRocks PS-7116

File: sql/opt_range.cc [hunk 2]
```diff
@@ -15074,6 +15074,8 @@ int QUICK_SKIP_SCAN_SELECT::get_next() {
   head->column_bitmaps_set_no_signal(&column_bitmap, head->write_set);
   do {
     if (!is_prefix_valid) {
+      head->file->set_end_range(NULL, handler::RANGE_SCAN_ASC);
```
Comment (george-lorch) [inline sql/opt_range.cc:15078]: Same, please add comment

**Review by george-lorch — APPROVED** [review]
LGTM by adding some comments

**Pull request merged**